From 0f42438bb7d1909294212aef3b768349fcf681ea Mon Sep 17 00:00:00 2001
From: Myles Watson <mylesgw@google.com>
Date: Thu, 4 Feb 2021 17:30:55 -0800
Subject: [PATCH 14/21] [Q_asb_2021-05] avrcp: Ignore AVCT commands that are
 too long

Bug: 177611958
Test: compilation
Tag: #security
Change-Id: If914b5928cdf16696eb54bfe91c2869cbbf8e36c
(cherry picked from commit dc07e927b1272be92095a47634dce0ba3b44b4fd)
---
 stack/avrc/avrc_api.cc | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/stack/avrc/avrc_api.cc b/stack/avrc/avrc_api.cc
index 15bab13360..a7f8d6426a 100644
--- a/stack/avrc/avrc_api.cc
+++ b/stack/avrc/avrc_api.cc
@@ -661,9 +661,10 @@ static void avrc_msg_cback(uint8_t handle, uint8_t label, uint8_t cr,
   tAVRC_MSG_VENDOR* p_msg = &msg.vendor;
 
   if (cr == AVCT_CMD && (p_pkt->layer_specific & AVCT_DATA_CTRL &&
-                         AVRC_PACKET_LEN < sizeof(p_pkt->len))) {
-    /* Ignore the invalid AV/C command frame */
-    p_drop_msg = "dropped - too long AV/C cmd frame size";
+                         p_pkt->len > AVRC_PACKET_LEN)) {
+    android_errorWriteLog(0x534e4554, "177611958");
+    AVRC_TRACE_WARNING("%s: Command length %d too long: must be at most %d",
+                       __func__, p_pkt->len, AVRC_PACKET_LEN);
     osi_free(p_pkt);
     return;
   }
-- 
2.35.1

